src/components/GitPanel.tsx

```diff
@@ -513,32 +571,30 @@ export class GitPanel extends React.Component<IGitPanelProps, IGitPanelState> {
             title: this.props.trans.__('Who is committing?'),
             body: new GitAuthorForm()
           });
+
           if (!result.button.accept) {
-            console.log('User refuses to set identity.');
-            return false;
-          }
-          const identity = result.value;
-          try {
-            await this.props.model.config({
-              'user.name': identity.name,
-              'user.email': identity.email
-            });
-          } catch (error) {
-            if (error instanceof Git.GitResponseError) {
-              console.log(error);
-              return false;
-            }
-            throw error;
+            throw new Error(
+              this.props.trans.__('User refused to set identity.')
+            );
           }
+
+          const { name, email } = result.value;
+          await this.props.model.config({
+            'user.name': name,
+            'user.email': email
+          });
         }
         this._previousRepoPath = path;
       } catch (error) {
+        if (error instanceof Git.GitResponseError) {
+          throw error;
+        }
+
         throw new Error(
           this.props.trans.__('Failed to set your identity. %1', error.message)
         );
       }
```

Comment:
I wasn't sure if checking for this specific Response Error was important or not. But I figured it was important since has that `traceback` property in the class definition which afaik would have gotten lost if I didn't check.